### PR TITLE
CI00151

### PR DESCRIPTION
**Pull request by erikzhang**
Close https://github.com/neo-project/neo-plugins/issues/113

### CODE REVIEW

File: RpcWallet/RpcWallet.cs [hunk 1]
```diff
@@ -82,17 +77,15 @@ public JObject OnProcess(HttpContext context, string method, JArray _params)
                             from = _params[0].AsString().ToScriptHash();
                             to_start = 1;
                         }
-                        JArray to = (JArray)_params[to_start + 0];
-                        long fee = _params.Count >= to_start + 2 ? long.Parse(_params[to_start + 1].AsString()) : 0;
```
Comment (shargon) [inline RpcWallet/RpcWallet.cs]: We should be able to attach ScopedSignatures, but maybe in other pull request

Comment (vncoelho) [reply RpcWallet/RpcWallet.cs]: Let's do in another PR, scoped was not merged, right?

File: ApplicationLogs/LogReader.cs [hunk 2]
```diff
@@ -78,7 +78,7 @@ public void OnPersist(Snapshot snapshot, IReadOnlyList<Blockchain.ApplicationExe
                     }
                     return notification;
                 }).ToArray();
-                writeBatch.Put(appExec.Transaction.Hash.ToArray(), json.ToString());
+                writeBatch.Put((appExec.Transaction.Hash ?? snapshot.PersistingBlock.Hash).ToArray(), json.ToString());
```
Comment (shargon) [inline ApplicationLogs/LogReader.cs]: Is not the same expected value

Comment (erikzhang) [reply ApplicationLogs/LogReader.cs]: Do you have any good ideas?

Comment (shargon) [reply ApplicationLogs/LogReader.cs]: In fact, is stored, but is not readed, but one extra property in the json should be great, something like 
```csharp
json["type"]=appExec.Transaction==null?"tx":"block";
```

File: ApplicationLogs/LogReader.cs [hunk 3]
```diff
@@ -52,7 +52,7 @@ public void OnPersist(Snapshot snapshot, IReadOnlyList<Blockchain.ApplicationExe
             foreach (var appExec in applicationExecutedList)
             {
                 JObject json = new JObject();
-                json["txid"] = appExec.Transaction.Hash.ToString();
+                json["txid"] = appExec.Transaction?.Hash.ToString();
```
Comment (shargon) [inline ApplicationLogs/LogReader.cs]:
```suggestion
                json["type"] = appExec.Transaction == null ? "block" : "tx";
                json["txid"] = appExec.Transaction?.Hash.ToString();
```

Comment (erikzhang) [reply ApplicationLogs/LogReader.cs]: We can check the `trigger` for knowing the type.

Comment (shargon) [reply ApplicationLogs/LogReader.cs]: The unique without Tx is System, right?

Comment (erikzhang) [reply ApplicationLogs/LogReader.cs]: Yes.

**Pull request merged**